### PR TITLE
feat(calendar): add methods for calendar

### PR DESCRIPTION
**Pull request by reb2020**
Moved methods from calendar to lib

### CODE REVIEW

File: src/calendar/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    return parseInt(offerPackageTimezoneOffset || 0)
+  }
+
+  return requestTimezoneOffset
+}
+
+const getMonthsToRequest = (timezoneOffset, maxDate) => {
+  const now = moment().utcOffset(timezoneOffset)
+  const diff = moment.duration(moment(maxDate).diff(now))
+  return diff.months() + (12 * diff.years()) + 2 // Add 1 month for current month, add 1 month for end month
```
Comment (Sonyque) [inline src/calendar/index.js:16]: How about?
```suggestion
return moment(maxDate).diff(now, 'months') + 2
```

Another question do we always need to add 1 month for the end month?

File: test/calendar/index.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    it('get return min date', () => {
+      expect(calendar.getStartDate('2022-01-30')).to.eql('2022-01-30')
+    })
+
+    it('get return travelFrom date', () => {
+      expect(calendar.getStartDate('2022-01-25', '2022-01-28')).to.eql('2022-01-28')
+    })
+
+    it('get return travelFrom date if non setup min date', () => {
+      expect(calendar.getStartDate(undefined, '2022-01-28')).to.eql('2022-01-28')
```
Comment (Sonyque) [inline test/calendar/index.test.js:32]: Could you also add a case of defined `minDate` and absent `travelFromDate`

File: src/calendar/index.js (hunk 1 above)
Comment (reb2020) [reply src/calendar/index.js:16]: Good question I just to move function from calendar

File: src/calendar/constants.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    dateFloorOffset: 2,
+    hourOfDayThreshold: 0,
+    useDynamicRate: false,
+    useTimezoneOffset: false,
+  },
+  [OFFER_TYPE_LAST_MINUTE_HOTEL]: {
+    type: OFFER_TYPE_LAST_MINUTE_HOTEL,
+    parentType: OFFER_TYPE_HOTEL,
+    dateFloorOffset: 0,
+    hourOfDayThreshold: 15,
```
[the remark this answers is not in the document]
Comment (reb2020) [reply src/calendar/constants.js:24]: Currently that responsible for different cases of code

File: test/calendar/index.test.js (hunk 2 above)
Comment (reb2020) [reply test/calendar/index.test.js:32]: That case already covered above 

```
it('get return min date', () => {
      expect(calendar.getStartDate('2022-01-30')).to.eql('2022-01-30')
    })
```

File: src/calendar/index.js (hunk 1 above)
Comment (reb2020) [reply src/calendar/index.js]: Updated function

**Review by Sonyque — APPROVED** [review]

**Pull request merged**